### PR TITLE
Merge test utility projects

### PR DESCRIPTION
**Pull request by emgarten**
* Moving test and sample apps out of the src directory.
* Moved SampleCommandLineExtensions under test\TestExtensions
* Cleaned up project references to remove transitive ones for some projects
* Created TestPackagesCore due to a conflict with a net45 class with the same name.
* Namespaces are left the same but could be changed so that all test helpers are in Test.Utility in the future.

Merging projects into *Test.Utility*
* NuGet.Test.Utility
* NuGet.Protocol.Test.Utility
* NuGet.Test.Server
* NuGet.Commands.Test.Utility
* NuGet.DependencyResolver.Core.Tests.Utility

*Test.Utility* is a common layer of test helpers that will sit above all product code, and below all actual test projects. There were many projects that were specific to a single product project containing just a few specific test helpers.

Merged synchronization tests into *NuGet.Common.Tests*
* NuGet.Synchronization.Test
* SynchronizationTestApp

SynchronizationTestApp has been removed, this was no longer used since moving from DNX To dotnet where it no longer worked. Locking testing is done through NuGet.Common.Tests already.

After the change VS shows 77 projects in the solution

//cc @rrelyea

### CODE REVIEW

File: test/NuGet.Core.Tests/NuGet.Common.Test/SynchronizationTest.cs [hunk 1]
```diff
@@ -418,7 +212,7 @@ public class SynchronizationTests
 
         private Task<int> WaitForInt1(CancellationToken token)
         {
-            int i = _value1;
+            var i = _value1;
```
Comment (mishra14) [inline test/NuGet.Core.Tests/NuGet.Common.Test/SynchronizationTest.cs:215]: 👍

File: test/NuGet.Core.Tests/NuGet.Common.Test/SynchronizationTest.cs [hunk 2]
```diff
@@ -229,147 +164,6 @@ public class SynchronizationTests
             Assert.False(timeout.IsCancellationRequested);
         }
 
-        [Fact(Skip = "Unable to make socket connection on the CLI")]
```
Comment (mishra14) [inline test/NuGet.Core.Tests/NuGet.Common.Test/SynchronizationTest.cs:232]: 👍

File: src/NuGet.Core/NuGet.Build.Tasks/Common/MSBuildUtility.cs [hunk 3]
```diff
@@ -9,7 +9,7 @@
 
 namespace NuGet.Build
 {
-    public static class MSBuildUtility
+    internal static class MSBuildUtility
```
Comment (mishra14) [inline src/NuGet.Core/NuGet.Build.Tasks/Common/MSBuildUtility.cs:12]: 👍

File: src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/Common/MSBuildTaskItem.cs [hunk 4]
```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
```
Comment (mishra14) [inline src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/Common/MSBuildTaskItem.cs:1]: Why were these 2 new files added?

**Review by mishra14 — COMMENTED** [review]
Commented

File: src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/Common/MSBuildTaskItem.cs (hunk 4 above)
Comment (emgarten) [reply src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/Common/MSBuildTaskItem.cs:1]: they were public and shared, which caused conflicts. only internal things should be shared.

**Review by mishra14 — APPROVED** [review]
discussed offline. looks good.

**Pull request merged**